### DISCUSSION TITLE
Prevent installation of certain packages

### DISCUSSION THREAD

**Discussion by jozefizso** [NuGet/Home#12808] [Q&A]
What tools does Nuget offer to prevent installation of certain packages?

We would like to be notified when installing new/updating existing packages if they reference some unwanted packages.

**Comment by kartheekp-ms** [comment]
> What tools does Nuget offer to prevent installation of certain packages?

When you say certain packages, do you mean vulnerable packages? If yes, [Auditing package dependencies for security vulnerabilities](https://learn.microsoft.com/nuget/concepts/auditing-packages) might help.

[dotnet list package](https://learn.microsoft.com/dotnet/core/tools/dotnet-list-package) command has several options to display deprecated and/or vulnerable packages including transitive dependencies.

[Package source mapping](https://learn.microsoft.com/nuget/consume-packages/package-source-mapping) might help to avoid downloading unwanted packages if the patterns are configured properly.

The [dotnet nuget trust command](https://learn.microsoft.com/dotnet/core/tools/dotnet-nuget-trust) manages the trusted signers. By default, NuGet accepts all authors and repositories. These commands allow you to specify only a specific subset of signers whose signatures will be accepted, while rejecting all others.
